client/src/components/Routes/AdminRoute.test.js

```diff
@@ -0,0 +1,111 @@
+import '@testing-library/jest-dom';
+import axios from 'axios';
+import { Outlet } from 'react-router-dom';
+import AdminRoute from './AdminRoute';
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import Spinner from "../Spinner";
+import { useAuth } from '../../context/auth';
+
+jest.mock('../../context/auth', () => ({
+    useAuth: jest.fn()
+}));
+
+jest.mock('mongoose', () => ({
+    connect: jest.fn(),
+    model: jest.fn(),
+    Schema: jest.fn(),
+}));
+
+// Sample token for testing
+const mockAuth = { token: 'sample token' };
+
+jest.mock('axios');
+
+jest.mock('react-router-dom', () => ({
+    Outlet: () => <div>Mocked Outlet</div>
+}));
+
+jest.mock('../Spinner', () => 
+    () => <div>Mocked Spinner</div>
+);
+
+describe('AdminRoute', () => {
+    beforeEach(() => {
+        jest.clearAllMocks();
+    });
+
+    it('should render Spinner if no auth token', async () => {
+        useAuth.mockReturnValue([null, jest.fn()]);
+        render(<AdminRoute />);
```

Comment:
Do you want to expect that `axios.get` is called with a particular argument, for example `/api/v1/auth/admin-auth`?